### PR TITLE
Enable documentation warnings and fix a few

### PR DESCRIPTION
**Pull request by tgoyne**
Turns on warnings for invalid documentation comments, removes a few bogus ones, and marks core headers as system headers to suppress warnings from them.

### CODE REVIEW

File: Realm.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1954,6 +1955,11 @@
 				MACOSX_DEPLOYMENT_TARGET = 10.9;
 				ONLY_ACTIVE_ARCH = YES;
 				OTHER_CFLAGS = "";
+				OTHER_CPLUSPLUSFLAGS = (
+					"$(OTHER_CFLAGS)",
+					"-isystem",
+					core/include,
+				);
```
Comment (bdash) [inline Realm.xcodeproj/project.pbxproj]: Do we still need `core/include` in `HEADER_SEARCH_PATHS` after doing this?

**Comment by segiddins** [comment]
Do those project file changes need to be mirrored in the podspec?

**Comment by jpsim** [comment]
:+1:

**Comment by tgoyne** [comment]
> Do those project file changes need to be mirrored in the podspec?

No point in turning on more warnings for the podspec, and at least in released versions the core headers should always be warning-clean on default settings.

**Comment by segiddins** [comment]
:+1:

**Pull request merged**